routemaster/models/client_token.rb

```diff
@@ -0,0 +1,37 @@
+require 'routemaster/models'
+require 'routemaster/mixins/redis'
+require 'securerandom'
+
+module Routemaster
+  module Models
+    class ClientToken
+      include Mixins::Redis
+
+      PREFIX = "api_keys:"
+
+      def self.get_all
+        key_dump = {}
+        existing_keys = _redis.keys "#{PREFIX}*"
+        nil if existing_keys.nil?
+        existing_keys.each do |key|
+          key_dump[key.gsub(PREFIX, '')] = _redis.keys key
+        end
+        key_dump
+      end
+
+      def self.generate_api_key(service_name)
+        unique_key = nil
+        while unique_key.nil? do
+          new_key = SecureRandom.uuid 
+          unique_key = new_key unless _redis.exists "#{PREFIX}#{new_key}"
+        end
```

Comment:
This loops feels unnecessary to me--the change of actually duplicating a duplicate UUID is essentially zero, especially as we're only generating a small number. I'd change it to use `SecureRandom.hex(16)` though as UUIDs are guaranteed to be _unique_ but are not guaranteed to be _unguessable_ so shouldn't really be used as security tokens (the Ruby implementation is based on a secure generator so the implementation _happens_ to be safe here, but it's not guaranteed; you also only get about 122 bits of entropy rather than 128 as part of the layout is fixed to the v4 identifier and a few other bits).